[Merged by Bors] - currency docs (#1305)
ComposableFi/composable: frame/assets-registry/README.md

## Basics

> Each remote asset must have a local identifier. This pallet calls [CurrencyFactory](../currency-factory/README.md) to get that done.

Reviewer: ```suggestion
Each remote asset must have a local identifier. This pallet uses [CurrencyFactory](../currency-factory/README.md) internally for that purpose.

```

## Decimals

> Remote assets may have different decimals than local ones, so remote assets may need to be configured to have proper decimals. 
> This mapping can be used by out-of-consensus protocols, like oracles and bridges.
> 
> When a transfer happens, we should know what the given `Amount` transferred means for that currency on our local network. We also need to know the minimal amount and number of decimals the currency has remotely.
> Wrong handling may lead to precision loss and loss of currency too. In the worst case, transfer back and forth will print money.
> 
> For example, remotely BTC can have eight decimals, while locally, we can have 12.

Reviewer: ```suggestion
Remote assets may have different decimals than local ones, so remote assets may need to be configured to have proper decimals. As an example, remotely BTC has eight decimals, while locally we use 12.

This mapping can be used by out-of-consensus protocols, such as oracles and bridges.

When a transfer happens, we should know what the given `Amount` transferred means for that currency on our local network. We also need to know the minimal amount and number of decimals the currency has remotely.

Mishandling may lead to precision loss and loss/gain of currency; in the worst case, a round-trip transfer will "print money".

```
